### PR TITLE
V2 search command branch

### LINKED ISSUE #12: Search for modules based on module level

### PR DESCRIPTION
**Pull request by deenliong**
fixed #12 #11 #10

### CODE REVIEW

**Review by kohnh — COMMENTED** [review]
IsSameModuleLevel has multiple return trues maybe you can link them together with || and your if else statement can be shortened to level.toString() == moduleLevel
In SearchModuleCommand class the FORMAT string can be removed if not used

File: src/main/java/seedu/duke/command/SearchModuleCommand.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+     * Filter the module list based on input module level.
+     *
+     * @param module the module object
+     * @param level  the level that user input
+     * @return true if module level matches input level
+     */
+    static boolean isSameModuleLevel(Module module, String level) {
+        // get the first int embedded in the module code
+        String moduleCode = module.moduleCode;
+        int moduleLevel = (Integer.parseInt(moduleCode.replaceAll("[^0-9]", ""))) / 1000;
```
Comment (owenl131) [inline src/main/java/seedu/duke/command/SearchModuleCommand.java:75]: can put the level calculation in model/Module.java

File: src/main/java/seedu/duke/command/SearchModuleCommand.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    /**
+     * Filters the module list based on the input semester.
+     *
+     * @param module           the module object
+     * @param toSearchSemester the semester that user input
+     * @return true if module is offered in the semester
+     */
+    static boolean isOfferedInSemester(Module module, String toSearchSemester) {
+        // convert toSearchSemester to int
+        int toSearchSemesterInt = Integer.parseInt(toSearchSemester);
```
Comment (owenl131) [inline src/main/java/seedu/duke/command/SearchModuleCommand.java:89]: can put some error handling

File: src/main/java/seedu/duke/command/SearchModuleCommand.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                }
+            }
+        }
+
+        // filter the searchResult if toSearchSemester is not empty and semester does not match
+        if (toSearchSemester != null) {
+            for (int i = 0; i < searchResult.size(); i++) {
+                if (!isOfferedInSemester(searchResult.get(i), toSearchSemester)) {
+                    searchResult.remove(i);
+                    i--;
```
Comment (owenl131) [inline src/main/java/seedu/duke/command/SearchModuleCommand.java:147]: can be a bit dangerous to modify a list while iterating over it, usually good practice is to create a new list and add the elements matching the criteria to the new list.

**Review by owenl131 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/duke/command/SearchModuleCommand.java (hunk 1 above)
Comment (deenliong) [reply src/main/java/seedu/duke/command/SearchModuleCommand.java:75]: shifted

File: src/main/java/seedu/duke/command/SearchModuleCommand.java (hunk 3 above)
Comment (deenliong) [reply src/main/java/seedu/duke/command/SearchModuleCommand.java]: updated to good practice

**Review by owenl131 — APPROVED** [review]
LGTM

**Pull request merged**